.svglintrc.js

```diff
@@ -163,6 +163,33 @@ module.exports = {
               }
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-path";
+
+            const iconPath = $.find("path").attr("d");
+            if (!updateIgnoreFile && isIgnored(reporter.name, iconPath)) {
+              return;
+            }
+
+            const { segments } = parsePath(iconPath);
+            const isSegmentInvalid = ([command, coord1, coord2, ...rest], index) => {
+              if (['m', 'l', 'h', 'v'].includes(command) && coord1 === 0 && coord2 === 0) {
+                return true;
+              }
+              if (index > 0) {
+                const [prevCoord2, prevCoord1, ...rest] = segments[index - 1].reverse();
+                if (['M', 'L', 'H', 'V'].includes(command) && coord1 === prevCoord1 && coord2 === prevCoord2) {
+                  return true;
+                }
+              }
+            };
+            if (segments.some(isSegmentInvalid)) {
+              reporter.error(`Unexpected segment in path.`);
+              if (updateIgnoreFile) {
+                ignoreIcon(reporter.name, iconPath, $);
+              }
+            }
```

Comment:
- We can improve the failure message reconstructing the command since all are made by one point.
- Also, we can show all failed commands (segments) for each failed path.
- Note that here I'm executing the `isSegmentInvalid` function only for segments that starts with possible invalid commands.
 - I've changed the "segment" word in the message replacing it with "command" because the SVG specification [describes them as commands](https://www.w3.org/TR/SVG/paths.html#Introduction).

```suggestion
            _invalidSegment = false;
            for (let s=0; s<segments.length; s++) {
              if (
                ['M', 'L', 'H', 'V', 'm', 'l', 'h', 'v'].includes(segments[s][0]) &&
                isSegmentInvalid(segments[s], s)
              ) {
                let readableSegment = `${segments[s][0]}${segments[s][1]} ${segments[s][2]}`;
                reporter.error(`Unexpected command ${readableSegment} in path.`);
                _invalidSegment = true;
              }
            }
            if (_invalidSegment && updateIgnoreFile) {
              ignoreIcon(reporter.name, iconPath, $);
            }
```

Reply:
Thanks for your review! I made some changes. Not exactly as you proposed, but inspired from. What do you think?

Reply:
Better than my proposal :+1: